### ISSUE TITLE
REST ProductImage Position is missing/null (impossible to know main or featured image)

### ISSUE THREAD

**Issue by alex254** [woothemes/woocommerce#31924]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

See https://woocommerce.github.io/woocommerce-rest-api-docs/#product-properties and then Product Image properties.
There is no property for image position returned on the rest api... This is really annoying as we can not know what is the featured image (position = 0)

### Expected behavior

return image position

### Actual behavior

property is not there

### Steps to reproduce

n/a

### WordPress Environment

rest api v3

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by woocommercebot** [comment]
Hi @alex254,

  Thank you for opening the issue! It requires further feedback from the WooCommerce Core team.

  We are adding the `needs developer feedback` label to this issue so that the Core team could take a look.

  Please note it may take a few days for them to get to this issue. Thank you for your patience.

**Comment by peterfabian** [comment]
Thanks for the request. 

It seems that the featured image is the first one in the response, but there's no way to tell if it's actually featured or it's the first one from the gallery, so I can see how that can be an issue.

I think we could solve this by e.g. adding a new property `featured image`, or `position`, as you say. I'd be actually leaning towards `featured image` as it's more semantic.

**Comment by the-csaba** [comment]
Hi all,

I'm starting to work on this. as part of the Contributor Day initiative.

**Comment by the-csaba** [comment]
Hello everyone,

While implementing this feature, I encountered a question and would appreciate your input before finalizing the changes.

Adding a new field for retrieving data is relatively straightforward, but I'm unsure about handling the create and update scenarios.

Currently, the first image in the images array is set as featured, while the rest are added to the gallery. However, with the new `featured` field, users can explicitly choose which image to set as featured (if any).

I understand that altering the REST API behaviour is not typically recommended, but we could implement this change in a backward-compatible manner.

Here's my proposed solution to address this issue:

1. We can maintain the existing behaviour if the `featured` field is absent from all image objects in the request.
2. If the `featured` field is present in all image objects, we respect the value and set the featured image accordingly.
3. If the `featured` field is absent from some image object, we set the featured image based on whether a `true` value exists; otherwise, we set the first image as featured.

Additionally, I have a related query:

If we introduce the `featured` field in the request, should we verify how many images have been set to true? Since only one image can be featured, we need to decide which one is set as featured when multiple true values exist. Or we can introduce a new 400 bad request response.

I would love to hear your thoughts on these questions.
